mods/cnc/maps/gdi09/gdi09.lua

```diff
@@ -0,0 +1,169 @@
+}
+
+AutoAttackWaves = { Nod1, Nod2, Nod3, Nod4, Nod5, Auto1, Auto2, Auto3, Auto4, Auto5, Auto6, Auto7, Auto8 }
+
+StationaryGuards = { Actor174, Actor173, Actor182, Actor183, Actor184, Actor185, Actor186, Actor187 , Actor199, Actor200, Actor201, Actor202, Actor203, Actor204}
+
+StartStationaryGuards = function()
+	Utils.Do(StationaryGuards, function(unit)
+		if not unit.IsDead then
+			unit.Patrol( { unit.Location } , true, 20)
+		end
+	end)
+end
+
+SendWaves = function(counter, Waves)
+	if counter <= #Waves then
+		local team = Waves[counter]
+
+		for type, amount in pairs(team.units) do
+			MoveAndHunt(Utils.Take(amount, Nod.GetActorsByType(type)), team.waypoints)
+		end
+
+		Trigger.AfterDelay(DateTime.Seconds(team.delay), function() SendWaves(counter + 1, Waves) end)
+	end
+end
+
+StartPatrols = function()
+	for k, team in pairs(Patrols) do
+		local group = 1
+		for type, amount in pairs(team.units) do
+			for i = 1, amount do
+				Reinforcements.Reinforce(Nod, { type }, { team.waypoints[team.initialWaypointPlacement[group]] }, 0, function(unit)
+					ReplenishPatrolUnit(unit, handofnod, team.waypoints, team.wait)
+				end)
+			end
+			group = group + 1
+		end
+	end
+	Patrols = nil
+end
+
+ReplenishPatrolUnit = function(unit, building, waypoints, waitatwaypoint)
+	unit.Patrol(waypoints, true, waitatwaypoint)
+	Trigger.OnKilled(unit, function()
+		local queueUnit = { unit = { unit.Type }, atbuilding = { building }, waypoints = waypoints }
+		PatrolProductionQueue[#PatrolProductionQueue + 1] = queueUnit
+	end)
+end
+
+SendGDIReinforcements = function()
+	Media.PlaySpeechNotification(GDI, "Reinforce")
+	Reinforcements.ReinforceWithTransport(GDI, InsertionHelicopterType, GDIHeliReinfUnits, { GDIHeliEntryNorth.Location, GDIHeliLZ.Location }, { GDIHeliLZ.Location + CVec.New(20, 0) })
+end
+
+SendGDIReinforcementChinook = function()
+	Reinforcements.ReinforceWithTransport(GDI, 'tran', nil, { GDIHeliEntryNorth.Location, GDIHeliLZ.Location })
+end
+
+SpawnGunboat = function()
+	Media.PlaySpeechNotification(GDI, "Reinforce")
+	Actor.Create("boat", true, { Owner = GDI, Facing = 0, Location = CPos.New(62,37) })
+end
+
+WorldLoaded = function()
+	GDI = Player.GetPlayer("GDI")
+	Nod = Player.GetPlayer("Nod")
+
+	Camera.Position = DefaultCameraPosition.CenterPosition
+
+	DestroyBunkers = GDI.AddObjective("Destroy the Nod bunkers to allow Carter's\nconvoy to pass through safely.")
+	Trigger.OnAllKilled(NodBunkersNorth, function()
+		GDI.MarkCompletedObjective(DestroyBunkers)
+		Trigger.AfterDelay(DateTime.Seconds(1), SpawnGunboat)
+	end)
+	Trigger.OnAllKilled(NodBunkersSouth, function()
+		GDI.MarkCompletedObjective(DestroyBunkers)
+		SendGDIReinforcementChinook()
+		Trigger.AfterDelay(DateTime.Seconds(1), SpawnGunboat)
+	end)
+	Trigger.OnEnteredFootprint({ CPos.New(60,37) }, function(a, id)
+		if a.type == "boat" then
```

Comment:
Try changing it to
```suggestion
		if a.Type == "boat" then
```

Reply:
I've changed it and tested a bit more.

A simple message  for debugging purpose
`Media.DisplayMessage("Actor " .. a.Type .. " entered TriggerFootprint")` 
before the `if a.Type == "boat"` check revealed that an LST does trigger OnEnteredFootprint while the gunboat doesn't. Is this a possible bug or a misconfig of my boat?

Reply:
The gunboat is a bit special case with the `TDGunboat` trait which does all the movement hacks. Maybe it doesn't trigger properly.